### PR TITLE
Make channel_contributions contain a channel axis and exploit that in plots

### LINKED ISSUE #118: MMMs should provide per channel/control contribution variables
At the moment, the MMM class [sums the contributions across channels](https://github.com/pymc-labs/pymc-marketing/blob/main/pymc_marketing/mmm/delayed_saturated_mmm.py#L98) and [across controls](https://github.com/pymc-labs/pymc-marketing/blob/main/pymc_marketing/mmm/delayed_saturated_mmm.py#L115). This makes it harder to plot contributions per channel/control since the curves have to be recomputed almost from scratch.
A better approach is to have `MMM`s guarantee that there will be a deterministic variable that tracks contributions across channels and time. This variable need not be limited to only 2 dimensions, but could also include region or brand, but the API should guarantee to have at least channels and time in it. That way, we can make plotting functions that are region/brand agnostic, and then reduce all potentially irrelevant things.

### PR DESCRIPTION
**Pull request by lucianopaz**
Closes #118 

This PR does two things.
1. It adds a `channel` and `control` dimension to the `channel_contributions` and `control_contributions` respectively. These deterministics are then used in the `BaseMMM` plotting methods.
2. The plotting functionality tests were moved to the `mmm.test_base.py` script and were made to run a proxy fitted toy MMM.

### CODE REVIEW

**Comment by lucianopaz** [comment]
This PR should also fix the flaky tests that popped up in #114

File: pymc_marketing/mmm/base.py [hunk 1]
```diff
@@ -231,48 +242,86 @@ def plot_posterior_predictive(
...
-            y=az.extract(self.fit_result, var_names=["intercept"]).mean().item(),
+            y=intercept.mean().data,
             color=f"C{i + 1}",
             ax=ax,
         )
         ax.fill_between(
             x=self.data_df[self.date_column],
             y1=intercept_hdi[:, 0],
-            y2=intercept_hdi[:, 1],
+            y2=intercept_hdi[:, 0],
```
Comment (juanitorduz) [inline pymc_marketing/mmm/base.py:324]: seems `y1` is the same as `y2` now? maybe this is not intentional?

Comment (lucianopaz) [reply pymc_marketing/mmm/base.py:324]: Oops, that was a typo

File: pymc_marketing/mmm/base.py [hunk 2]
```diff
@@ -88,6 +88,14 @@ def preprocessing_methods(self) -> List[Callable[[pd.DataFrame], pd.DataFrame]]:
             if getattr(method, "_tags", {}).get("preprocessing", False)
         ]
 
+    def get_target_transformer(self):
+        try:
+            return self.target_transformer
+        except AttributeError:
+            null_scaler = MaxAbsScaler()
+            null_scaler.fit([[0], [1]])
```
Comment (juanitorduz) [inline pymc_marketing/mmm/base.py:96]: Is this to create a "trivial" scaler?

Comment (lucianopaz) [reply pymc_marketing/mmm/base.py:96]: Yes. I wanted a scaler that did nothing. Is there a better way?

File: tests/mmm/test_base.py [hunk 3]
```diff
@@ -30,6 +33,123 @@
...
+        def build_model(self, data_df, **kwargs):
+            pass
+
+    mmm = ToyMMM(
+        toy_df,
+        target_column="y",
+        date_column="date",
+        channel_columns=["channel_1", "channel_2"],
+    )
+    mmm.control_columns = ["control_1", "control_2"]
```
Comment (juanitorduz) [inline tests/mmm/test_base.py:48]: Can we also test when there are no control variables ?

File: pymc_marketing/mmm/base.py (hunk 2 above)
Comment (juanitorduz) [reply pymc_marketing/mmm/base.py:96]: I think this is ok! I wonder if is an overkill just adding a `IdentityTransformer` just for the sake of readability...

**Comment by juanitorduz** [comment]
The changes are quite nice! specially regarding the better use of `xarray` API. I once the plotting tests also handle the "con conrtol variables" case  we are good to go :)

**Comment by lucianopaz** [comment]
I discovered that the failing tests were due to a bug in `matplotlib` that was caused by a change introduced to `numpy` 1.24. The [matplotlib bug](https://github.com/matplotlib/matplotlib/issues/24106) was fixed in version 3.6.1, but seaborn still hasn't made a release that is compatible with that version. At the moment, the only thing that we can do to avoid this bug is to downgrade numpy in the requirements. I'll add an issue that keeps track of this, so that when seaborn cuts a release, we can unpin numpy and everything else.

File: tests/mmm/test_base.py (hunk 3 above)
Comment (lucianopaz) [reply tests/mmm/test_base.py]: Done

File: pymc_marketing/mmm/base.py (hunk 2 above)
Comment (lucianopaz) [reply pymc_marketing/mmm/base.py]: I don't see any `IdentityTransformer` in the sklearn documentation. From what I saw, it looks like a [`FunctionTransformer`](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.FunctionTransformer.html#sklearn.preprocessing.FunctionTransformer) with `func=None` would do the job.

Comment (juanitorduz) [reply pymc_marketing/mmm/base.py]: As you wish :) No strong opinions here.

**Comment by juanitorduz** [comment]
> I discovered that the failing tests were due to a bug in `matplotlib` that was caused by a change introduced to `numpy` 1.24. The [matplotlib bug](https://github.com/matplotlib/matplotlib/issues/24106) was fixed in version 3.6.1, but seaborn still hasn't made a release that is compatible with that version. At the moment, the only thing that we can do to avoid this bug is to downgrade numpy in the requirements. I'll add an issue that keeps track of this, so that when seaborn cuts a release, we can unpin numpy and everything else.

Thanks for documenting this!

File: pymc_marketing/mmm/base.py [hunk 4]
```diff
@@ -88,6 +88,13 @@ def preprocessing_methods(self) -> List[Callable[[pd.DataFrame], pd.DataFrame]]:
             if getattr(method, "_tags", {}).get("preprocessing", False)
         ]
 
+    def get_target_transformer(self):
```
Comment (juanitorduz) [inline pymc_marketing/mmm/base.py:91]: add return type hint?

**Review by juanitorduz — APPROVED** [review]
LGTM 👍 (just a small type hint comment)

File: pymc_marketing/mmm/base.py (hunk 4 above)
Comment (lucianopaz) [reply pymc_marketing/mmm/base.py]: Done
